lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.ts

```diff
@@ -282,43 +339,62 @@ export class ContentMetadataComponent implements OnChanges, OnInit, OnDestroy {
-
-        return properties.length > 0;
+    private loadProperties(node: Node) {
+        if (node) {
+            this.basicProperties$ = this.getProperties(node);
+            this.groupedProperties$ = this.contentMetadataService.getGroupedProperties(node, this.preset);
+            if (this.displayTags) {
+                this.loadTagsForNode(node.id);
+            }
+            this.loadCategoriesForNode(node.id);
+            if (!this.node.aspectNames.includes('generalclassifiable')) {
+                this.categories = [];
+                this.classifiableChangedSubject.next();
+            }
+        }
     }
 
-    ngOnDestroy() {
-        this.onDestroy$.next(true);
-        this.onDestroy$.complete();
+    private getProperties(node: Node) {
+        const properties$ = this.contentMetadataService.getBasicProperties(node);
+        const contentTypeProperty$ = this.contentMetadataService.getContentTypeProperty(node);
+        return zip(properties$, contentTypeProperty$)
+            .pipe(map(([properties, contentTypeProperty]) => {
+                const filteredProperties = contentTypeProperty.filter((property) => properties.findIndex((baseProperty) => baseProperty.key === property.key) === -1);
+                return [...properties, ...filteredProperties];
+            }));
     }
 
-    public canExpandTheCard(group: CardViewGroup): boolean {
-        return group.title === this.displayAspect;
+    private isEmpty(value: any): boolean {
+        return value === undefined || value === null || value === '';
     }
 
-    public canExpandProperties(): boolean {
-        return !this.expanded || this.displayAspect === 'Properties';
+    private loadCategoriesForNode(nodeId: string) {
+        this.assignedCategories = [];
+        this.categoryService.getCategoryLinksForNode(nodeId).subscribe((categoryPaging) => {
+            this.categories = categoryPaging.list.entries.map((categoryEntry) => categoryEntry.entry);
+            this.assignedCategories = [...this.categories];
+        });
     }
 
-    keyDown(event: KeyboardEvent) {
-        if (event.keyCode === 37 || event.keyCode === 39) { // ArrowLeft && ArrowRight
-            event.stopPropagation();
+    private saveCategories(): { [key: string]: Observable<CategoryPaging | CategoryEntry | void> } {
+        const observables: { [key: string]: Observable<CategoryPaging | CategoryEntry | void> } = {};
+        if (this.categories) {
+            this.assignedCategories.forEach((assignedCategory) => {
+                if (this.categories.every((category) => category.name !== assignedCategory.name)) {
+                    observables[`Removing ${assignedCategory.id}`] = this.categoryService.unlinkNodeFromCategory(this.node.id, assignedCategory.id);
+                }
+            });
+            const categoryLinkBodies: CategoryLinkBody[] = [];
+            this.categories.forEach((category) => {
```

Comment:
This can be shorter, something like:

```ts
const categoryLinkBodies = this.categories.map((category) => {
    const categoryLinkBody = new CategoryLinkBody();
    categoryLinkBody.categoryId = category.id;
    return category;
});
```

Thanks that you can initialize and declare categoryLinkBodies in the same line (you don't need even initialize that with empty array) and you don't need push categories manually to array.

Reply:
Shortened